packages/components/src/schemas/main.ts

```diff
@@ -0,0 +1,36 @@
+import type { TSchema } from "@sinclair/typebox"
+
+import type { IsomerComponentTypes } from "~/types"
+import { IsomerPageSchema } from "~/types"
+import {
+  IsomerComplexComponentsMap,
+  IsomerNativeComponentsMap,
+} from "./components"
+
+const definitions = {
+  components: {
+    complex: IsomerComplexComponentsMap,
+    native: IsomerNativeComponentsMap,
+  },
+}
+
+export const schema: TSchema = {
+  $schema: "http://json-schema.org/draft-07/schema#",
+  title: "Isomer Next Page JSON",
+  ...IsomerPageSchema,
+  ...definitions,
+}
+
+export const getComponentSchema = (
+  component: IsomerComponentTypes,
+): TSchema => {
+  const componentSchema =
+    component === "prose"
```

Comment:
We check only for prose, as prose comprises of all the native components right

Reply:
Yup, so it will always be complex + 'prose' so we treat prose as a special kind of component.